src/routes/(app)/select-namespace/+page.svelte

```diff
@@ -16,32 +16,27 @@
 
   import type { DescribeNamespaceResponse as Namespace } from '$types';
 
-  $: namespaceNames = $namespaces.map(
-    (namespace: Namespace) => namespace?.namespaceInfo?.name,
-  );
+  let searchValue = '';
 
-  $: namespaceList = namespaceNames.map((namespace: string) => {
-    return {
-      namespace,
-      onClick: async (namespace: string) => {
-        const { authorized } = await fetchWorkflowForAuthorization(namespace);
-        if (authorized) {
-          $lastUsedNamespace = namespace;
-          goto(routeForWorkflows({ namespace }));
-        } else {
-          toaster.push({
-            variant: 'error',
-            message: translate('namespaces.unauthorized-namespace-error'),
-          });
-        }
-      },
-    };
-  });
+  const navigateToNamespace = async (namespace: string) => {
+    const { authorized } = await fetchWorkflowForAuthorization(namespace);
+    if (authorized) {
+      $lastUsedNamespace = namespace;
+      goto(routeForWorkflows({ namespace }));
+    } else {
+      toaster.push({
+        variant: 'error',
+        message: translate('namespaces.unauthorized-namespace-error'),
+      });
+    }
+  };
 
-  let searchValue = '';
-  $: filteredList = namespaceList.filter(({ namespace }) =>
-    namespace.includes(searchValue),
-  );
+  $: filteredList = $namespaces
+    .map((namespace: Namespace, index: number) => ({
+      namespace: namespace?.namespaceInfo?.name,
+      index,
+    }))
+    .filter(({ namespace }) => namespace.includes(searchValue));
```

Comment:
Is separating these two potentially a little more performant since we wouldn't be iterating over the list twice on search?

```suggestion
  $: namespaceList = $namespaces.map((namespace: Namespace, index: number) => ({
    namespace: namespace?.namespaceInfo?.name,
    index,
  }));
  $: filteredList = namespaceList.filter(({ namespace }) =>
    namespace.includes(searchValue),
  );
```

Reply:
I really wish the `VirtualList` would give me the index and then I wouldn't have to even map over the namespaces at all.